core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -88,23 +88,23 @@ private class ShuffleStatus(numPartitions: Int) {
    * Register a map output. If there is already a registered location for the map output then it
    * will be replaced by the new location.
    */
-  def addMapOutput(mapId: Int, status: MapStatus): Unit = synchronized {
-    if (mapStatuses(mapId) == null) {
+  def addMapOutput(mapIndex: Int, status: MapStatus): Unit = synchronized {
```

Comment:
This is a place that I think `mapIndex` makes sense.

Reply:
Thanks for pointing this out.